posts/2023-08-22-ArgoCD-Configuration-Drift.adoc

```diff
@@ -0,0 +1,156 @@
+// // // // // // // //
+
+== Install the Open Liberty Operator ==
+The Open Liberty Operator aids in deploying and managing Open Liberty Applications on Kubernetes. Our example deployment makes use of the Open Liberty Operator to deploy the sample application. If you prefer to deploy using pure Kubernetes deployment files, this step is optional.
+
+To install it with the default configuration of watching all namespaces, clone the repository and run the following commands:
+```
+git clone https://github.com/OpenLiberty/open-liberty-operator.git
+kubectl create namespace open-liberty
+kubectl apply --server-side -k open-liberty-operator/deploy/releases/1.2.2/kustomize/overlays/watch-all-namespaces
+```
+
+== Install ArgoCD ==
+To install ArgoCD simply run these commands from the Quickstart section of the docs:
+```
+kubectl create namespace argocd
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+```
+This will create a new namespace named argocd as well as apply the install manifests to that namespace.  It alternatively can be installed via OLM.
+
+
+== Using ArgoCD ==
+Get the default Admin password using the following command:
+```
+kubectl get secret argocd-initial-admin-secret -n argocd -o jsonpath='{.data.password}' | base64 -d
+```
+Next, go to the route of the `argocd-server` and log with the `admin` user and password.
+[.img_border_light]
+image::/img/blog/configuration-drift-1.png[Argo Login,width=70%,align="center"]
+
+Once logged, in you should see this page:
+[.img_border_light]
+image::/img/blog/configuration-drift-2.png[Argo Home,width=70%,align="center"]
+
+== Creating a repository to be used for GitOps ==
+Now that you've successfully installed ArgoCD and gained access to its console, the next step is to set up a GitHub repository for Argo CD to synchronize with.
+
+For the purposes of this blog, we'll use the Daytrader7 sample application. This application is built around the concept of an online stock trading system, and the Daytrader7 README provides a comprehensive overview.
+
+It's a recommended best practice to maintain separation between configuration repositories and code repositories. Keeping these separate ensures that infrastructure changes are decoupled from application changes, allowing for more granular control, better audit trails, and reduced risk of unintended side-effects. For this example, we'll create a deploy directory within our existing repository to house the deployment configuration.
+
+To begin, navigate to GitHub and fork this repository: https://github.com/WASdev/sample.daytrader7. Use your forked repository when configuring ArgoCD in the following steps
+
+We assume that your CI pipleline has a way to build contianer image for your application. For this example, we have a pre-built cotnainer image used in the deployment files `deploy/daytrader7-deploy.yaml`.
+
+== Deploying DayTrader 7 via GitOps ==
```

Comment:
Can make this subtitle more broad so it appeals to more devs. So... "Deploying a Java application with GitOps"